pkg/flow/internal/controller/loader.go

```diff
@@ -144,27 +120,26 @@ func (l *Loader) Apply(parentScope *vm.Scope, blocks []*ast.BlockStmt, configBlo
 			components = append(components, c)
 			componentIDs = append(componentIDs, c.ID())
 
-			if err = l.evaluate(logger, parentScope, c); err != nil {
+			if err = l.evaluateComponent(logger, parentScope, c); err != nil {
 				var evalDiags diag.Diagnostics
 				if errors.As(err, &evalDiags) {
 					diags = append(diags, evalDiags...)
 				} else {
 					diags.Add(diag.Diagnostic{
 						Severity: diag.SeverityLevelError,
 						Message:  fmt.Sprintf("Failed to build component: %s", err),
-						StartPos: ast.StartPos(n.(*ComponentNode).block).Position(),
-						EndPos:   ast.EndPos(n.(*ComponentNode).block).Position(),
+						StartPos: ast.StartPos(n.(*ComponentNode).Block()).Position(),
+						EndPos:   ast.EndPos(n.(*ComponentNode).Block()).Position(),
```

Comment:
Looks like you can simplify this even further, since `c` is in scope as a `*ComponentNode`, and is an alias to `n`:

```suggestion
						StartPos: ast.StartPos(c.Block()).Position(),
						EndPos:   ast.EndPos(c.Block()).Position(),
```